### PR TITLE
feat: ebics client

### PR DESCRIPTION
**Pull request by barredterra**
Adds a new doctype **Ebics User**, which represents a natural person's bank access. It holds the IDs provided by the bank. An **Ebics User** is located in a certain **Country** and belongs to a **Company** and to a **Bank**. It has access to one or more bank accounts.

<details>
<summary>EBICS User</summary>

ebics_user

</details>

<details>
<summary>Rearranged Banking Settings</summary>

bs_klarna

bs_ebics

</details>

Todo:

- [x] ~~allow sync from **Banking Settings**, iterate over **Ebics User**~~ Link to **Ebics User** from **Banking Settings**
- [x] use a separate signature passphrase, store keyring and sig_passphrase in **Ebics User**
- [ ] Show banner in **Ebics User** if last sync was unsuccessful, with specific error message. Not sure how to test this.
- [x] Display Bank account limit in Subscription Details
- [x] validate unique UserID / PartnerID 
- [x] rename "Bank Account Limit" to "Ebics User Limit"
- [x] Hide unnecessary "Actions" (Initialize, Verify) when not in developer mode
- [x] Fetch license key via API
- [x] Transition: Two checkboxes "Enable Kosma", "Enable Ebics". Both can be active at the same time.
- [ ] Once the first Ebics User is activated show message to disable Kosma.
- [x] Make storing the passphrase optional?

### CODE REVIEW

File: banking/ebics/doctype/ebics_user/ebics_user.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		Admin().request.remove_ebics_user(host_id, self.partner_id, self.user_id)
+
+	def validate_country_code(self):
+		country_code = frappe.db.get_value("Country", self.country, "code")
+		if not country_code or len(country_code) != 2:
+			frappe.throw(
+				_("Please add a two-letter country code to country {0}").format(
+					get_link_to_form("Country", self.country)
+				)
+			)
```
Comment (marination) [inline banking/ebics/doctype/ebics_user/ebics_user.py:71]: we can make sure the country code is mandatory in frappe ? Are there cases where this asserts?

**Review by marination — CHANGES_REQUESTED** [review]
API requests happening on EBICS User save() and on "Initialize" etc. need error handling

Otherwise, looks great. What's pending is the flow/UX review

File: banking/ebics/doctype/ebics_user/ebics_user.py (hunk 1 above)
Comment (barredterra) [reply banking/ebics/doctype/ebics_user/ebics_user.py:71]: https://github.com/frappe/frappe/pull/28307

File: banking/ebics/doctype/ebics_user/ebics_user.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	user = frappe.get_doc("EBICS User", ebics_user)
+	user.check_permission("write")
+
+	manager = get_ebics_manager(user)
+	manager.activate_bank_keys()
+	user.db_set("bank_keys_activated", 1)
+
+
+@frappe.whitelist()
+def download_bank_statements(
```
Comment (marination) [inline banking/ebics/doctype/ebics_user/ebics_user.py:158]: Enqueue this in case of a time consuming process ?

**Comment by barredterra** [comment]
@Mergifyio backport version-14

**Pull request merged**